pkg/sql/schemachanger/scplan/internal/rules/dep_drop_constraint.go

```diff
@@ -18,31 +18,34 @@ import (
 
 // These rules ensure that constraint-dependent elements, like an constraint's
 // name, etc. disappear once the constraint reaches a suitable state.
+// TODO (xiang): The dep rules here are not complete, as they are aimed specifically
+// for check constraints only. Complete them when we properly support the
+// other two constraints: UniqueWithIndex and ForeignKey.
```

Comment:
Actually there's foreign keys, index-backed uniques, and also UniqueWithoutIndex as well as not-NULL.